libcudacxx/.upstream-tests/test/CMakeLists.txt

```diff
@@ -120,13 +120,16 @@ add_subdirectory(host_only)
 
 find_program(libcudacxx_LIT lit REQUIRED)
 
+set(libcudacxx_LIT_FLAGS "" CACHE STRING "Flags for libcudacxx's lit test suite.")
+message(STATUS "libcudacxx_LIT_FLAGS: ${libcudacxx_LIT_FLAGS}")
+
 if (NOT LIBCUDACXX_TEST_WITH_NVRTC)
   # Build but don't run the tests. Used by CI to pre-seed sccache for the test machines.
   # Only executed if explicitly requested.
   add_custom_target(libcudacxx.test.lit.precompile
     "${CMAKE_COMMAND}" -E env
       "LIBCUDACXX_SITE_CONFIG=${lit_site_cfg_path}"
-    "${libcudacxx_LIT}" -vv --no-progress-bar
+    "${libcudacxx_LIT}" -vv --no-progress-bar "${libcudacxx_LIT_FLAGS}"
```

Comment:
@allisonvacanti @robertmaynard I'm driving myself crazy trying to figure out what I'm doing wrong here, but this isn't currently working because when I try to do:
```
./ci/build_libcudacxx.sh -cmake-options "-Dlibcudacxx_LIT_FLAGS='-s --time-tests'"
```

it expands to:
```
-s\ --time-tests
```

which fails. 

It seems cmake is escaping the space in the string. I've tried about a dozen different ways of doing this and I can't get any of them to work. 

plz help.

Reply:
You should be able to omit the quotes around `${libcudacxx_LIT_FLAGS}`, since you'll want the individual flags parsed as single options. If any args in that variable need quotes, they should be added when defining it.

Reply:
Try playing with the quotes here:

```
./ci/build_libcudacxx.sh -cmake-options "-Dlibcudacxx_LIT_FLAGS='-s --time-tests'"
```

I think what we want is something like:

```
./ci/build_libcudacxx.sh -cmake-options "\"-Dlibcudacxx_LIT_FLAGS=-s --time-tests\""
```

Reply:
(and if more `-D` options are needed, add them outside of the escaped quotes -- those are just there to associate the `LIT_FLAGS` value with the option)

./ci/build_libcudacxx.sh -cmake-options "\"-Dlibcudacxx_LIT_FLAGS=-s --time-tests\" -DFOO=BAR \"-DBAZ=Another value with spaces...\""